### PR TITLE
Add ability to author float initializers

### PR DESCRIPTION
**Pull request by baijumeswani**
This pull request adds a way to author float initializers. Example usage:

```py
class WeightedAvg(model_type):
    def __init__(self, w1, w2):
        super(WeightedAvg, self).__init__()

        self.loss1 = onnxblock.loss.CrossEntropyLoss()
        self.loss2 = onnxblock.loss.CrossEntropyLoss()
        self.w1 = onnxblock.building_blocks.Constant(w1)
        self.w2 = onnxblock.building_blocks.Constant(w2)
        self.mul = onnxblock.building_blocks.Mul()
        self.add = onnxblock.building_blocks.Add()

    def build(self, loss_input_name1, loss_input_name2):
        return self.add(
            self.mul(self.w1(), self.loss1(loss_input_name1, labels_name="labels1")),
            self.mul(self.w2(), self.loss2(loss_input_name2, labels_name="labels2")),
        )

weighted_model = WeightedAvg(4.0, 1.5)
with onnxblock.onnx_model(onnx_model):
    weighted_averaged_output = weighted_model(loss_input1, loss_input2)
```

### CODE REVIEW

**Comment by ashbhandare** [comment]
What are all the accepted types by the Value() class- eg Python lists, tuple, numpy arrays etc?

**Comment by baijumeswani** [comment]
For now, only floats. But the implementation can be modified over time to create any kind of initializer we like.

File: orttraining/orttraining/python/training/onnxblock/building_blocks.py [hunk 1]
```diff
@@ -178,3 +179,19 @@ def __init__(self):
 class Neg(_UnaryOp):
     def __init__(self):
         super(Neg, self).__init__("Neg")
+
+
+class Value(Block):
```
Comment (pengwa) [inline orttraining/orttraining/python/training/onnxblock/building_blocks.py:184]: in which case do we need this? just for during some loss function building?

Looks they will not be trainable param, right? so why we call it weight?

**Comment by thiagocrepaldi** [comment]
> This pull request adds a way to author float initializers. Example usage:
> 
> ```python
> class WeightedAvg(model_type):
>     def __init__(self, w1, w2):
>         super(WeightedAvg, self).__init__()
> 
>         self.loss1 = onnxblock.loss.CrossEntropyLoss()
>         self.loss2 = onnxblock.loss.CrossEntropyLoss()
>         self.w1 = onnxblock.building_blocks.Value(w1)
>         self.w2 = onnxblock.building_blocks.Value(w2)
>         self.mul = onnxblock.building_blocks.Mul()
>         self.add = onnxblock.building_blocks.Add()
> 
>     def build(self, loss_input_name1, loss_input_name2):
>         return self.add(
>             self.mul(self.w1(), self.loss1(loss_input_name1, labels_name="labels1")),
>             self.mul(self.w2(), self.loss2(loss_input_name2, labels_name="labels2")),
>         )
> 
> weighted_model = WeightedAvg(4.0, 1.5)
> with onnxblock.onnx_model(onnx_model):
>     weighted_averaged_output = weighted_model(loss_input1, loss_input2)
> ```

what is authroing an initializer?

**Comment by baijumeswani** [comment]
@thiagocrepaldi authoring initializers refers to a way for users to add initializers to the onnx model with ease. These initializers can be floats, ints, and even numpy arrays (which will get converted to a TensorProto in the onnx model initializer). This PR only supports doing that for floats for now.

File: orttraining/orttraining/python/training/onnxblock/building_blocks.py (hunk 1 above)
Comment (baijumeswani) [reply orttraining/orttraining/python/training/onnxblock/building_blocks.py:184]: One of the user model has two outputs. The training model is built by combining these two outputs as weighted average:
```py
output = w1*celoss(output0)+w2*celoss(output1)
```

For the user to be able to do this, we need a way for them to define what w1 and w2 is. Calling it weight only in the unit tests which uses the initializers as weighted average coefficients.

**Comment by baijumeswani** [comment]
> Does "Value" means they are constant values? If yes, maybe we can call them Constant to align with ONNX constant operator.

Yeah, Value means a constant in the onnx model (an initializer). My future goal is to represent every input/output as a `Value` type and override the `+, -, /, *` operators so that users can do `output = 4*celoss(out0)+2.5*celoss(out1). Which is why I called this Value.

But you're right, we should call it Constant for the time being and enhance it over time. Will make the change.

**Review by pengwa — APPROVED** [review]
Better to resolve or ignore the pylint errors.

**Comment by ashbhandare** [comment]
> For now, only floats. But the implementation can be modified over time to create any kind of initializer we like.

Can we add documentation around this in the code?

**Pull request merged**